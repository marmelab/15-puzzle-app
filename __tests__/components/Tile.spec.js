import 'react-native';
import React from 'react';
import Tile from '../../src/components/Tile';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('should render Tile correctly', () => {
    renderer.create(<Tile onPress={() => {}} tileValue={10} />);
});

it('should match snapshot', () => {
    const tree = renderer
        .create(<Tile onPress={() => {}} tileValue={12} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
