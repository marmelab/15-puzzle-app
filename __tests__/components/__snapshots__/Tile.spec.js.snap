// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#d32f2f",
          "display": "flex",
          "height": 60,
          "justifyContent": "center",
          "margin": 4,
          "width": 60,
        },
        Object {
          "elevation": 3,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    12
  </Text>
</View>
`;
